test/functions.php

```diff
@@ -40,5 +40,72 @@ public function testMalformeddisplayCharacters()
         
     }
    
+    public function testSuccevalidateFields()
+    {
+        //expected result of the test
+        $expected = true;
+        //input for the test to get the result
+        $testInput1 = ['charname' => 'Lilen', 'class' => 'Sorcerer', 'level' => '6', 'strength' => 8, 'dexterity' => 12, 'constitution' => 14, 'intelligence' => 13, 'wisdom' => 12, 'charisma' => 18];
+        //run the real function with the input
+        $case = validateFields($testInput1);
+        //compare the expected result with the actual result
+        $this->assertEquals($expected, $case);
+    }
+
+    public function testFailurevalidateFields()
+    {
+        //expected result of the test
+        $expected = false;
+        //input for the test to get the result
+        $testInput1 = ['charname' => 'Lilen', 'class' => 'Sorcerer', 'level' => '', 'strength' => 12, 'dexterity' => 12, 'constitution' => 14, 'intelligence' => 13, 'wisdom' => 12, 'charisma' => 18];
+        //run the real function with the input
+        $case = validateFields($testInput1);
+        //compare the expected result with the actual result
+        $this->assertEquals($expected, $case);
```

Comment:
There is a slightly more elegant way to do this, there is actually built in assertion called assertFalse()

So you could ditch your $expected, and switch your assertion to `$this->assertFalse($case);`